client/src/model/geo_slice.ts

```diff
@@ -0,0 +1,54 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { ApiClient } from '@/api/api_client';
+import { GeoIdentifiers } from '@/model/geo_identifiers';
+import { AppDispatch } from '@/model/store';
+
+/**
+ * State of geo selection.
+ */
+interface GeoState {
+  geoids?: GeoIdentifiers;
+}
+
+const initialState: GeoState = {};
+const client = new ApiClient();
+
+const geoSlice = createSlice({
+  name: 'geoSlice',
+  initialState,
+  reducers: {
+    geoIdsFetchedSuccess(state: GeoState, action: PayloadAction<GeoIdentifiers>) {
+      state.geoids = { ...state.geoids, ...action.payload };
+    },
+    geoIdsFetchedError(state: GeoState, action) {
+      // TODO(breuch): Consider showing user message.
+      console.log(`Error fetching geos: ${state} ${action}`);
+    },
+    userQueriedGeo(state: GeoState, action: PayloadAction<GeoIdentifiers>) {
```

Comment:
whats the difference between this and geoIdsFetchedSuccess / why do we call this before calling API in action below?

Reply:
To set the stage for a "loading" state. We may want to do some UX to show we are trying to retrieve data (esp for users with slow connections)